subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -316,8 +316,9 @@ uint8_t ll_adv_aux_sr_data_set(uint8_t handle, uint8_t op, uint8_t frag_pref,
 	/* Check if data will fit in remaining space */
 	/* TODO: need aux_chain_ind support */
 	ext_hdr_len = sr_dptr - &sr_com_hdr->ext_hdr_adv_data[0];
-	if ((PDU_AC_EXT_HEADER_SIZE_MIN + ext_hdr_len + len) >
-	    PDU_AC_PAYLOAD_SIZE_MAX) {
+	if ((len > CONFIG_BT_CTLR_ADV_DATA_LEN_MAX) ||
+	    ((PDU_AC_EXT_HEADER_SIZE_MIN + ext_hdr_len + len) >
+	     PDU_AC_PAYLOAD_SIZE_MAX)) {
```

Comment:
Hmm, you are changing 3 places that follows this pattern:

```c
if (len > MAX_A || (b + len) > MAX_B)
```

Does it make sense to add a macro or function for this check? 
If not, fine to keep as is

Reply:
This code would change as part of long AD data support that is work in progress wherein total AD data length will have to be compared with `CONFIG_BT_CTLR_ADV_DATA_LEN_MAX` and the particular PDUs AD data fragment len with `PDU_AC_PAYLOAD_SIZE_MAX`. We can revisit use of macro then.